common/src/main/java/com/kumuluz/ee/configuration/utils/ConfigurationUtil.java

```diff
@@ -153,11 +156,19 @@ public void set(String key, Float value) {
         config.getConfigurationSources().get(0).set(key, value);
     }
 
-    public void subscribe(ConfigurationListener listener) {
+    public void subscribe(ConfigurationListener listener, String key) {
+
         config.getDispatcher().subscribe(listener);
+
+        watch(key);
+
     }
 
     public void unsubscribe(ConfigurationListener listener) {
         config.getDispatcher().unsubscribe(listener);
     }
+
+    public void notifyChange(String key, String value) {
```

Comment:
Remove this method in the util in order to not allow anyone to publish changes to the config. Instead pass the ConfigurationDispatcher instance to the ConfigurationSource and make sure the source calls notify on that instance.